lib/node/nodes/merge.js

```diff
@@ -23,59 +23,72 @@ var Merge = Node.create(TYPE, PARAMS);
 
 module.exports = Merge;
 
-var INPUT_LEFT_ALIAS = '_cdb_analysis_left_source';
-var INPUT_RIGHT_ALIAS = '_cdb_analysis_right_source';
-
 var JOIN_OPERATIONS = {
     inner: 'INNER JOIN',
     left: 'LEFT JOIN',
     right: 'RIGHT JOIN'
 };
 
 Merge.prototype.sql = function() {
-    var columns = this._getColumns();
+    var buster = utilId.buster();
+
+    var left_alias = '_cdb_analysis_left_source' + buster;
+    var right_alias = '_cdb_analysis_right_source' + buster;
+
+    var columns = this._getColumns(left_alias, right_alias);
 
     var sql = queryTemplate({
         columns: columns.join(', '),
-        input_left_alias: INPUT_LEFT_ALIAS,
-        input_right_alias: INPUT_RIGHT_ALIAS,
+        input_left_alias: left_alias,
+        input_right_alias: right_alias,
         input_left: this.left_source.getQuery(),
         join_operation: JOIN_OPERATIONS[this.join_operator],
         input_right: this.right_source.getQuery(),
-        input_left_column_join_on: INPUT_LEFT_ALIAS + '.' + this.left_source_column,
-        input_right_column_join_on: INPUT_RIGHT_ALIAS + '.' + this.right_source_column
+        input_left_column_join_on: left_alias + '.' + this.left_source_column,
+        input_right_column_join_on: right_alias + '.' + this.right_source_column
     });
 
     debug(sql);
 
     return sql;
 };
 
-Merge.prototype._getColumns = function() {
-    var geomColumn = ((this.source_geometry === 'left_source') ? INPUT_LEFT_ALIAS : INPUT_RIGHT_ALIAS) +
+Merge.prototype._getColumns = function(left_alias, right_alias) {
+    var geomColumn = ((this.source_geometry === 'left_source') ? left_alias : right_alias) +
         '.the_geom as the_geom';
 
     var columns = [geomColumn]
-        .concat(this._getLeftColumns())
-        .concat(this._getRightColumns());
+        .concat(this._getLeftColumns(left_alias))
+        .concat(this._getRightColumns(right_alias));
 
     return columns;
 };
 
-Merge.prototype._getLeftColumns = function() {
+Merge.prototype._getLeftColumns = function(left_alias) {
     var leftColumns = (this.left_source_columns === null) ?
         this.left_source.getColumns(true) :
         this.left_source_columns;
 
-    leftColumns = setAliasTo(leftColumns, INPUT_LEFT_ALIAS, 'left');
+    if (this.source_geometry === 'left_source') {
+        leftColumns = leftColumns.filter(function (column) {
+            return (/^.*the_geom$/.test(column) === false);
```

Comment:
I would extract this function to a `named function` and reuse it in the `_getRightColumns` function:

```javascript
function isNotAGeometryColumn(column) {
    return (/^.*the_geom$/.test(column) === false);
});
```

and then use it like:

```javascript
if (this.source_geometry === 'right_source') {
    rightColumns = rightColumns.filter(isNotAGeometryColumn);
}
```

Reply:
Nice. I'll also change it to `endsWith` for clarity